File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseBetween(ListNode* head, int left, int right) {
        
        if (!head || !head->next || left == right) {
            return head;
        }
        
        ListNode *dummy = new ListNode(0, head);
        
        ListNode *current = dummy;
        
        
        for(int i = 0; i < left-1; i++) {
            current = current->next;
        }
        
        
        ListNode *prev = current;
        current = current->next;
        
        for(int i = left; i < right; i++) {
            ListNode *next = current->next;
            current->next = next->next;
            next->next = prev->next;
            prev->next = next;
        }
        
        return dummy->next;  
        
        
        
        
        
     
        
        
        
    }
};

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        
        if head is None or head.next is None or left == right:
            return head
        
        dummy = ListNode(0, head)
        
        current = dummy
        
        for i in range(left-1):
            current = current.next
        
        prev = current
        current = current.next
        
        for i in range(left, right):
            nxt = current.next
            current.next = nxt.next
            nxt.next = prev.next
            prev.next = nxt
        
        
        return dummy.next